code/datums/martial/monk.dm

```diff
@@ -0,0 +1,455 @@
+			D.visible_message("<span class='danger'>[A] punches [D], and they start to quiver!</span>", \
+					  "<span class='userdanger'>[A] punches you, and you feel your heart stop!</span>")
+	else
+		D.visible_message("<span class='danger'>[A] missed [D]!</span>", \
+				  "<span class='userdanger'>[A] misses you!</span>")
+	return 1
+
+/datum/martial_art/monk/grab_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(check_streak(A,D))
+		return 1
+	..()
+
+/datum/martial_art/monk/disarm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(check_streak(A,D))
+		return 1
+	..()
+
+/datum/action/monk_rest
+	name = "Rest - Sleep to regain your strength."
+	button_icon_state = "neckchop" // todo: replace
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+
+/datum/action/monk_rest/Trigger()
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't use [name] while you're incapacitated.</span>")
+		return
+	var/mob/living/carbon/human/H = owner
+	H.AdjustUnconscious(200)
+	to_chat(H, "<span class='warning'>You fall into a meditative sleep...</span>")
+	var/datum/martial_art/monk/MA = H.mind.martial_art
+	for(var/ME in MA.available_actions)
+		var/datum/action/monk/M = ME
+		M.uses_left = M.max_uses
+
+/datum/action/flurry_toggle
+	name = "Flurry Of Blows - Enable/Disable, attack more in 1 turn but with an attack penalty at early levels."
+	button_icon_state = "neckchop" // todo: replace
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	var/datum/martial_art/monk/martial
+
+/datum/action/flurry_toggle/Trigger()
+	if(martial)
+		martial.using_flurry = !martial.using_flurry
+		if(martial.using_flurry)
+			to_chat(owner, "<span class='warning'>You enable Flurry of Blows.</span>")
+		else
+			to_chat(owner, "<span class='warning'>You disable Flurry of Blows.</span>")
+
+/datum/action/monk
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	var/uses_left
+	var/max_uses
+	var/skill_name
+	var/pretty_name
+	var/datum/martial_art/monk/martial
+
+/datum/action/monk/Trigger()
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't use [name] while you're incapacitated.</span>")
+		return
+	if(!uses_left)
+		to_chat(owner, "<span class='warning'>You must rest before you can use [pretty_name]!</span>")
+		return
+	var/mob/living/carbon/human/H = owner
+	if (H.mind.martial_art.streak == skill_name)
+		owner.visible_message("<span class='danger'>[owner] assumes a neutral stance.</span>", "<b><i>Your next attack is cleared.</i></b>")
+		H.mind.martial_art.streak = ""
+	else
+		owner.visible_message("<span class='danger'>[owner] prepares a technique!</span>", "<b><i>Your next attack will be a [pretty_name].</i></b>")
+		H.mind.martial_art.streak = skill_name
+
+
+/datum/action/monk/stunning_fist
+	name = "Stunning Fist - A disabling strike that stuns for 3 seconds."
+	button_icon_state = "neckchop" // todo: replace
+	uses_left = 4
+	max_uses = 4
+	skill_name = "stunning_fist"

```

Comment:
Link this to type. Magic strings big gay.

Reply:
Will try to figure out how.

Reply:
Think this is the only meaningful issue left I believe, I'll see if I can get some help or figure it out on my own.

Reply:
I believe I did what you wanted, however I'm not sure. Put another comment here if this isn't.